### ISSUE TITLE
.Net: Bug: PromptExecutionSettings are frozen and cannot be modified.

### ISSUE THREAD

**Issue by shuaihuadu** [microsoft/semantic-kernel#6752]
**Describe the bug**
When I was using Kernel.CreateFunctionFromPrompt, I set parameters through HuggingFacePromptExecutionSettings, and an exception was thrown:
`System.InvalidOperationException: PromptExecutionSettings are frozen and cannot be modified.`

**To Reproduce**

Copy the code below into `Connectors.HuggingFace.UnitTests\HuggingFaceKernelBuilderExtensionsTests.cs` in the  sk-dotnet project, and then run the unit test.

```csharp
[Fact]
public async Task RunKernelFunctionWithHuggingFaceShouldRunCorrectlyAsync()
{
    Kernel kernel = Kernel.CreateBuilder()
        .AddHuggingFaceChatCompletion("model", new Uri("http://localhost:8080"))
        .Build();

    var kernelFunction = kernel.CreateFunctionFromPrompt("Write a C# Hello world", new HuggingFacePromptExecutionSettings
    {
        Temperature = 0.4f,
        TopP = 1
    });

    await foreach (var update in kernel.InvokeStreamingAsync<StreamingChatMessageContent>(kernelFunction))
    {
        if (update.Content is { Length: > 0 })
        {
            Console.Write(update.Content);
        }
    }
}
```

**Expected behavior**
The kernel function run success and return correctly  results.

**Screenshots**
![image](https://github.com/microsoft/semantic-kernel/assets/17045801/62f31a5f-c716-4331-b7a3-1c75343a4f6e)

**Platform**
 - OS: Windows 11
 - IDE: Visual Studio
 - Language: C#
 - Source:  The sk-dotnet source code.

**Comment by markwallace-microsoft** [comment]
@shuaihuadu I have a fix here: https://github.com/microsoft/semantic-kernel/pull/6895. Please take a look at the tests that are included with the fix.

**Comment by shuaihuadu** [comment]
> @shuaihuadu I have a fix here: #6895. Please take a look at the tests that are included with the fix.

Thanks for fixing this!
